apollo-api/src/jvmMain/kotlin/com/apollographql/apollo/api/ScalarType.kt

```diff
@@ -0,0 +1,10 @@
+package com.apollographql.apollo.api
+
+/**
+ * Represents a custom GraphQL scalar type
+ */
+actual interface ScalarType {
+  actual fun typeName(): String
+
+  fun javaType(): Class<*>
```

Comment:
I'm not 100% sure about this. Right now, this is the only choice to keep it compatible. This means, `javaType` will work as before for Jvm/Android.

Right now, this is also used by generated code. So, it would be blocker moving forward to figure this out.

Reply:
We need to think about this, this is used only to locate default scalar type adapter for system types like Int, Boolean, String etc.

Reply:
This will be done in a follow-up by also changing the code generation to accomodate.